### ISSUE TITLE
RUN <<EOF without an interpreter and cachi2

### ISSUE THREAD

**Issue by owtaylor** [konflux-ci/build-definitions#1200]
```dockerfile
RUN <<EOF
mkdir -p /etc/foo
cp -aR /src/foo /etc/foo
EOF
```

gets turned into:

```dockerfile
RUN . /cachi2/cachi2.env && <<EOF
mkdir -p /etc/foo
cp -aR /src/foo /etc/foo
EOF
```

which is weirdly not a syntax error, but a no-op. The right transformation would be something like:

```dockerfile
RUN . /cachi2/cachi2.env && /bin/sh <<EOF
mkdir -p /etc/foo
cp -aR /src/foo /etc/foo
EOF
```

Once you consider `RUN --mount=type=bin,src=/a,dest=/b <<EOF` wedging this into the current code gets a little tricky.

**Comment by chmeliik** [comment]
@owtaylor I would assume this is not critical since there is a fairly easy workaround (use `/bin/sh <<EOF` rather than just `<<EOF`)?

But we should fix this at some point. A good Dockerfile parser+editor will be needed.

**Comment by chmeliik** [comment]
Just came across https://www.mankier.com/1/cstrans-df-run (in https://github.com/kdudka/konflux-test-multicluster-global-hub/pull/9/commits/88998934c0b4ffeb9a333316ac89037dbfe60963)

It doesn't handle heredocs correctly either, but perhaps a contribution would be reasonable (it's a C++ codebase, haven't looked into it much)

**Comment by cgwalters** [comment]
> which is weirdly not a syntax error, but a no-op. 

Yeah, man I got burned by this bad. It took me quite a while to debug. I guess in bash it's something like passing the heredoc to `&&` which just discards it? Mind blowing...really seems like it should be a syntax error since there's no reason to ever do that.

> Once you consider RUN --mount=type=bin,src=/a,dest=/b <<EOF wedging this into the current code gets a little tricky.

Yes, which we also do.

**Comment by chmeliik** [comment]
This is fixed. If your prefetch-dependencies task is at least version 0.3.1, and your buildah task at least 0.10.3, then the build no longer modifies RUN instructions but instead sets env vars with `--secret` + `--mount`. This supports bare heredocs and exec-form instructions as well.

This change was missing in the changelog, currently being added in https://github.com/konflux-ci/build-definitions/pull/3649
